### PR TITLE
feat: adding alert component with markdown

### PR DESCRIPTION
**Pull request by JessicaSachs**
## Summary
* Adding an Alert component that is expandable, dismissible, or static.
* Removing the Markdown.vue component in favor of `useMarkdown`.
* Refactoring `Warnings.vue` to use the markdown composable.
* Creating the `useExternalLink` composable for use with `ExternalLink.vue` and within the `useMarkdown` composable.

### Alert component
Screen Shot 2021-12-01 at 1 43 03 AM
Screen Shot 2021-12-01 at 1 43 52 AM

```js
<Alert title="My title" icon="MyIcon" dismissible>Content</Alert>
```

### useMarkdown Composable

```js
const target = ref()
const text = '**Hello** world, `this is a code block'
const markdown = useMarkdown(target, text, { classes: { code: 'bg-pink-200 text-pink-500' } })
```

```html
<div ref="target" v-html="markdown" />
```

### useExternalLink Composable
Used by useMarkdown Composable to linkify anything with an href (conditionally, by default).

#### Use with href
```js
const open = useExternalLink('https://google.com')

open() // Executes the gql mutation to open the user's browser
```

#### Pass in href
```js
const open = useExternalLink()
open('https://google.com')
```

### CODE REVIEW

File: packages/frontend-shared/src/components/Alert.spec.tsx [hunk 1]
```diff
@@ -0,0 +1,223 @@
...
+    })
+  })
+})
+
+describe('playground', () => {
+  it('renders', () => {
+    cy.mount(() => (
+      <div class="p-4 text-center space-y-2">
+        <Select/>
+        <hr/>
```
Comment (JessicaSachs) [inline packages/frontend-shared/src/components/Alert.spec.tsx:211]: Kill or finish filling in

File: .vscode/settings.json [hunk 2]
```diff
@@ -21,5 +21,9 @@
   "editor.codeActionsOnSave": {
     "source.fixAll.eslint": true
   },
-  "typescript.tsdk": "node_modules/typescript/lib"
-}
\ No newline at end of file
+  "typescript.tsdk": "node_modules/typescript/lib",
+  "i18n-ally.keystyle": "nested",
+  "cSpell.words": [
+    "composables"
```
Comment (JessicaSachs) [inline .vscode/settings.json:27]: Move to cspell.json

File: .vscode/settings.json [hunk 3]
```diff
@@ -21,5 +21,9 @@
   "editor.codeActionsOnSave": {
     "source.fixAll.eslint": true
   },
-  "typescript.tsdk": "node_modules/typescript/lib"
-}
\ No newline at end of file
+  "typescript.tsdk": "node_modules/typescript/lib",
+  "i18n-ally.keystyle": "nested",
```
Comment (JessicaSachs) [inline .vscode/settings.json:25]: kill

File: .vscode/extensions.json [hunk 4]
```diff
@@ -25,7 +25,7 @@
 		// Description: An extension for setting-up multiple terminals at once, or just running some commands
 		// There are several Terminals defined in `.vscode/terminals.json` that can be used via this plugin.
 		"fabiospampinato.vscode-terminals",
-		// Name: Test Utils
+  // Name: Test Utils
 		// Description: Add, remove, and move .only in tests
 		"chrisbreiding.test-utils",
 		// Name: Toggle Quotes
```
Comment (JessicaSachs) [inline .vscode/extensions.json:31]: Bad merge conflict. Also, should be using spaces not tabs.

File: packages/frontend-shared/src/components/AlertHeader.vue [hunk 5]
```diff
@@ -0,0 +1,69 @@
...
+  title?: string
+  prefixIcon?: FunctionalComponent<SVGAttributes, {}> | null
+  suffixIcon?: FunctionalComponent<SVGAttributes, {}> | null
+  suffixIconAriaLabel?: string
+  alertClass?: string
+  prefixIconClass?: string
+  suffixIconClass?: string
+  headerClass?: string
+  suffixButtonClass?: string
+}>(), { title: 'Alert' })
```
Comment (JessicaSachs) [inline packages/frontend-shared/src/components/AlertHeader.vue:64]: I was getting local linter errors for this not having default values. Unsure if you are too.

File: .vscode/settings.json (hunk 2 above)
Comment (elevatebart) [reply .vscode/settings.json]: Moved to cspell with merge

File: .vscode/settings.json (hunk 3 above)
Comment (elevatebart) [reply .vscode/settings.json]: merged with the rest on merge commit

File: .vscode/extensions.json (hunk 4 above)
Comment (elevatebart) [reply .vscode/extensions.json]: Fixed

File: packages/frontend-shared/src/components/AlertHeader.vue (hunk 5 above)
Comment (elevatebart) [reply packages/frontend-shared/src/components/AlertHeader.vue:64]: I did too, I disabled the rule

File: packages/frontend-shared/src/components/Alert.spec.tsx (hunk 1 above)
Comment (elevatebart) [reply packages/frontend-shared/src/components/Alert.spec.tsx]: Killed...

**Review by elevatebart — DISMISSED** [review]

File: patches/@toycode+markdown-it-class+1.2.4.patch [hunk 6]
```diff
@@ -0,0 +1,13 @@
+diff --git a/node_modules/@toycode/markdown-it-class/index.js b/node_modules/@toycode/markdown-it-class/index.js
+index 8e8c8f4..5e2539d 100644
+--- a/node_modules/@toycode/markdown-it-class/index.js
++++ b/node_modules/@toycode/markdown-it-class/index.js
+@@ -8,7 +8,7 @@ const toArray = a => (Array.isArray(a) ? a : [a])
+ 
+ function parseTokens(tokens) {
+   tokens.forEach(token => {
+-    if (/(_open$|image)/.test(token.type) && mapping[token.tag]) {
```
Comment (lmiller1990) [inline patches/@toycode+markdown-it-class+1.2.4.patch:9]: wooooo lol

**Review by lmiller1990 — DISMISSED** [review]

File: patches/@toycode+markdown-it-class+1.2.4.patch (hunk 6 above)
Comment (JessicaSachs) [reply patches/@toycode+markdown-it-class+1.2.4.patch:9]: lol

**Pull request merged**